include/ddc/kernels/splines/spline_builder.hpp

```diff
@@ -81,12 +81,13 @@ class SplineBuilder
     // interpolator specific
     std::unique_ptr<ddc::detail::Matrix> matrix;
 
+    double m_dx; // average cell size for normalization of derivatives
```

Comment:
Non-const variable shouldn't be public. This variable can be const.
```suggestion
    double const m_dx; // average cell size for normalization of derivatives
```
But I wonder if it would make sense for `SplineBuilderBatched` to inherit from this class rather than storing an instance?

Reply:
SplineBuilder will disappear from ddc at some point. I just wait the moment when everything related to splines will be ddc-based to do so (avoiding severe break in the middle of the GPU port)

Reply:
Actually i would prefer non const variables inside classes even if they don't get modified after initialization. The reason is that they don't get moved correctly if they are const. I agree that for a scalar it does not make a difference but then we end up with a lot of rules...

Reply:
Even for public variables?!

Reply:
No i would try to make it private, is it required to be public ?

Reply:
There's no good reason to have public variables, if you need to expose it, do it through an accessor